plugins/outputs/postgresql/template/template.go

```diff
@@ -0,0 +1,429 @@
+   ]
+
+A simple example for usage with TimescaleDB would be:
+ [outputs.postgresql]
+   create_templates = [
+     '''CREATE TABLE {{ .table }} ({{ .allColumns }})''',
+     '''SELECT create_hypertable({{ .table|quoteLiteral }}, 'time', chunk_time_interval => INTERVAL '1d')''',
+     '''ALTER TABLE {{ .table }} SET (timescaledb.compress, timescaledb.compress_segmentby = 'tag_id')''',
+     '''SELECT add_compression_policy({{ .table|quoteLiteral }}, INTERVAL '2h')''',
+   ]
+...where the defaults for the other templates would be automatically applied.
+
+A very complex example for versions of TimescaleDB which don't support adding columns to compressed hypertables (v<2.1.0), using views and unions to emulate the functionality, would be:
+ [outputs.postgresql]
+   schema = "telegraf"
+   create_templates = [
+     '''CREATE TABLE {{ .table }} ({{ .allColumns }})''',
+     '''SELECT create_hypertable({{ .table|quoteLiteral }}, 'time', chunk_time_interval => INTERVAL '1d')''',
+     '''ALTER TABLE {{ .table }} SET (timescaledb.compress, timescaledb.compress_segmentby = 'tag_id')''',
+     '''SELECT add_compression_policy({{ .table|quoteLiteral }}, INTERVAL '2d')''',
+     '''CREATE VIEW {{ .table.WithSuffix "_data" }} AS
+          SELECT {{ .allColumns.Selectors | join "," }} FROM {{ .table }}''',
+     '''CREATE VIEW {{ .table.WithSchema "public" }} AS
+          SELECT time, {{ (.tagTable.Columns.Tags.Concat .allColumns.Fields).Identifiers | join "," }}
+          FROM {{ .table.WithSuffix "_data" }} t, {{ .tagTable }} tt
+          WHERE t.tag_id = tt.tag_id''',
+   ]
+   add_column_templates = [
+     '''ALTER TABLE {{ .table }} RENAME TO {{ (.table.WithSuffix "_" .table.Columns.Hash).WithSchema "" }}''',
+     '''ALTER VIEW {{ .table.WithSuffix "_data" }} RENAME TO {{ (.table.WithSuffix "_" .table.Columns.Hash "_data").WithSchema "" }}''',
+     '''DROP VIEW {{ .table.WithSchema "public" }}''',
+
+     '''CREATE TABLE {{ .table }} ({{ .allColumns }})''',
+     '''SELECT create_hypertable({{ .table|quoteLiteral }}, 'time', chunk_time_interval => INTERVAL '1d')''',
+     '''ALTER TABLE {{ .table }} SET (timescaledb.compress, timescaledb.compress_segmentby = 'tag_id')''',
+     '''SELECT add_compression_policy({{ .table|quoteLiteral }}, INTERVAL '2d')''',
+     '''CREATE VIEW {{ .table.WithSuffix "_data" }} AS
+          SELECT {{ .allColumns.Selectors | join "," }}
+          FROM {{ .table }}
+          UNION ALL
+          SELECT {{ (.allColumns.Union .table.Columns).Selectors | join "," }}
+          FROM {{ .table.WithSuffix "_" .table.Columns.Hash "_data" }}''',
+     '''CREATE VIEW {{ .table.WithSchema "public" }}
+          AS SELECT time, {{ (.tagTable.Columns.Tags.Concat .allColumns.Fields).Identifiers | join "," }}
+          FROM {{ .table.WithSuffix "_data" }} t, {{ .tagTable }} tt
+          WHERE t.tag_id = tt.tag_id''',
+   ]
+*/
+package template
```

Comment:
Maybe name it `sqltemplate` to make the linter happy?

Reply:
Guessing you mean this:

```
import-shadowing: The name 'template' shadows an import name
```
I'm not sure why golangci-lint is complaining about that. The statement it's making is wrong. There is no shadowing going on. But sure, I can rename.